src/Lykke.Service.HFT/Middleware/DoubleGreaterThanZeroAttribute.cs

```diff
@@ -1,28 +1,20 @@
-﻿using System.ComponentModel.DataAnnotations;
+﻿using Lykke.Service.HFT.Core.Validation;
+using Lykke.Service.HFT.Properties;
+using System;
+using System.ComponentModel.DataAnnotations;
 
 namespace Lykke.Service.HFT.Middleware
 {
-    public class DoubleGreaterThanZeroAttribute : ValidationAttribute
+    [AttributeUsage(AttributeTargets.Property | AttributeTargets.Field)]
+    public sealed class DoubleGreaterThanZeroAttribute : ValidationAttribute
     {
-        public override string FormatErrorMessage(string name)
+        public DoubleGreaterThanZeroAttribute()
         {
-            return string.Format("{0} must be more than zero.", name);
+            ErrorMessageResourceType = typeof(Resources);
+            ErrorMessageResourceName = nameof(Resources.ValueMustBePositive);
         }
 
         public override bool IsValid(object value)
-        {
-            return value as double? > double.Epsilon;
-        }
-
-        protected override ValidationResult IsValid(
-            object value, ValidationContext validationContext)
-        {
-            if (IsValid(value))
-                return ValidationResult.Success;
-            else
-                return new ValidationResult(
-                    FormatErrorMessage(validationContext.MemberName)
-                );
-        }
+            => !(value is double price) || PriceValidationRule.IsValid(price);
```

Comment:
Absolutely not critical, but this attribute is about checking any (double) properties, not just 'price'.

Reply:
renamed